sql/hive/src/main/scala/org/apache/spark/sql/hive/test/TestHive.scala

```diff
@@ -51,6 +51,11 @@ object TestHive
         // SPARK-8910
         .set("spark.ui.enabled", "false")))
 
+trait TestHiveSingleton {
+  protected def sqlContext: SQLContext = TestHive
+  protected def hiveContext: HiveContext = TestHive
```

Comment:
these can technically be `val`s, in which case subclasses can just `import hiveContext.implicits._` directly if they want.